File: Serene/Serene.Script/Administration/Language/LanguageGrid.cs

namespace Serene.Administration
{
    using jQueryApi;
    using Serenity;
    using System.Runtime.CompilerServices;

    [Imported]
    public class LanguageGrid : EntityGrid<LanguageRow>
    {
        public LanguageGrid(jQueryObject container)
            : base(container)
        {
        }
    }

    /* 
    This class has been ported to TypeScript. See LanguageGrid.ts
    Code below is only a reference for those who want to use Saltaralle

    [ColumnsKey("Administration.Language"), IdProperty("Id"), NameProperty("LanguageName")]
    [DialogType(typeof(LanguageDialog)), LocalTextPrefix("Administration.Language"), Service("Administration/Language")]
    public class LanguageGrid : EntityGrid<LanguageRow>, IAsyncInit
    {
        public LanguageGrid(jQueryObject container)
            : base(container)
        {
        }
    }
    */
}

File: Serene/Serene.Core/Modules/Administration/Language/LanguageGrid.ts
namespace Serene.Administration {

    @Serenity.Decorators.registerClass()
    export class LanguageGrid extends Serenity.EntityGrid<LanguageRow, any> {
        protected getColumnsKey() { return "Administration.Language"; }
        protected getDialogType() { return LanguageDialog; }
        protected getIdProperty() { return LanguageRow.idProperty; }
        protected getLocalTextPrefix() { return LanguageRow.localTextPrefix; }
        protected getService() { return LanguageService.baseUrl; }

        constructor(container: JQuery) {
            super(container);
        }

        protected getDefaultSortBy() {
            return [LanguageRow.Fields.LanguageName];
        }
    }
}